src/coreclr/tools/superpmi/superpmi/jithost.cpp

```diff
@@ -7,21 +7,17 @@
 #include "icorjitinfo.h"
 #include "jithost.h"
 
-// Look for 'key' as an environment variable named COMPlus_<key>. The returned value
-// is nullptr if it is not found, or a string if found. If not nullptr, the returned
-// value must be freed with jitInstance.freeLongLivedArray(value).
-WCHAR* GetCOMPlusVariable(const WCHAR* key, JitInstance& jitInstance)
-{
-    static const WCHAR Prefix[]  = W("COMPlus_");
-    static const size_t  PrefixLen = (sizeof(Prefix) / sizeof(Prefix[0])) - 1;
+#include <clrconfignocache.h>
```

Comment:
As a nit, this perhaps should be in runtimedetails.h, but I guess this is ok.

Reply:
Yup, it's an existing header, which I have reused (to avoid redefining consts and lengths).

@AaronRobinsonMSFT, ideally we could have a unified API with flag-y options to avoid duplicate functions. The usage of such API in jisthost.cpp would look like:

> `WCHAR* envVar = ClrConfig::Get(key, &returnBuffer, (ClrConfig_USEPREFIX & ~ClrConfig_USECACHE));`

Having different classes for cache vs. nochace is somewhat verbose.

Reply:
> @AaronRobinsonMSFT, ideally we could have a unified API with flag-y options to avoid duplicate functions. The usage of such API in jisthost.cpp would look like:

Sure, but that doesn't change the reason that the nocache version exists. Note that the nocache doesn't do any memory management nor does it work with `WCHAR`, it is entirely for `char`. This was done due to the various places that can't include or accept the lifetime management nor the requirement of chaining in the PAL. We can move the nocache, but it has broader build constraints about where it is used and done in the way it is to reduce that duplication.